interchain/scripts/src/bin/full_deploy.rs

```diff
@@ -50,6 +41,14 @@ fn full_deploy(mut networks: Vec<ChainInfoOwned>) -> anyhow::Result<()> {
                 monarch: monarch.to_string(),
             },
         )?;
+
+        // We register the gas asset of the chain to make sure there is a base asset there
+        let asset_denom = network.gas_denom;
```

Comment:
Not sure about the logic here, that wouldn't work for archway>arch for instance

But it's kind of hard to get a denom name from a raw denom.

One solution would be to leverage the cosmos off-chain chain registry